src/EnergyPlus/HVACUnitarySystem.cc

```diff
@@ -2586,40 +2619,50 @@ namespace HVACUnitarySystem {
                                   0.0,
                                   0.0,
                                   0.0); // conduct the sizing operation in the VS WSHP
-            UnitarySystem(UnitarySysNum).NumOfSpeedCooling = VarSpeedCoil(UnitarySystem(UnitarySysNum).CoolingCoilIndex).NumOfSpeeds;
+
+            if (NumSpeeds != VarSpeedCoil(UnitarySystem(UnitarySysNum).CoolingCoilIndex).NumOfSpeeds) {
+                ShowWarningError(RoutineName + ": " + CompType + " = " + CompName);
+                ShowContinueError("Number of cooling speeds does not match coil object.");
+                ShowFatalError("Cooling coil = " + VarSpeedCoil(UnitarySystem(UnitarySysNum).CoolingCoilIndex).VarSpeedCoilType + ": " + VarSpeedCoil(UnitarySystem(UnitarySysNum).CoolingCoilIndex).Name);
+            }
             DXCoolCap = GetCoilCapacityVariableSpeed(
                 cAllCoilTypes(UnitarySystem(UnitarySysNum).CoolingCoilType_Num), UnitarySystem(UnitarySysNum).CoolingCoilName, ErrFound);
             EqSizing.DesCoolingLoad = DXCoolCap;
 
-            for (Iter = 1; Iter <= UnitarySystem(UnitarySysNum).NumOfSpeedCooling; ++Iter) {
+            for (Iter = NumSpeeds; Iter >= 1;
+                 --Iter) { // use reverse order since we divide by CoolVolumeFlowRate(max)
                 UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(Iter) =
                     VarSpeedCoil(UnitarySystem(UnitarySysNum).CoolingCoilIndex).MSRatedAirVolFlowRate(Iter);
                 UnitarySystem(UnitarySysNum).CoolMassFlowRate(Iter) = UnitarySystem(UnitarySysNum).CoolVolumeFlowRate(Iter) * StdRhoAir;
-                // it seems the ratio should reference the actual flow rates, not the fan flow ???
```

Comment:
So here, we set speed ratio relative to the fan if we knew what the fan flow rate was. That seems correct to me. Otherwise we set = coil max flow which is incorrect if fan flow is different from coil flow.